tests/jms-tests/src/test/java/org/apache/activemq/artemis/jms/tests/ConnectionTest.java

```diff
@@ -65,6 +67,33 @@ public void testManyConnections() throws Exception
    //
 
    @Test
+   public void testThroughNewConnectionFactory() throws Exception
+   {
+      testThroughNewConnectionFactory(new ActiveMQConnectionFactory());
+      testThroughNewConnectionFactory(new ActiveMQConnectionFactory("tcp://localhost:61616"));
+      testThroughNewConnectionFactory(new ActiveMQConnectionFactory("tcp://localhost:61616", "guest", "guest"));
+   }
+
+   private void testThroughNewConnectionFactory(ActiveMQConnectionFactory factory) throws Exception
+   {
+      Connection conn = factory.createConnection();
+      conn.close();
+
+      try (JMSContext ctx = factory.createContext())
+      {
+         ctx.createProducer().send(ctx.createQueue("queue"),"Test");
+      }
+
+      try (JMSContext ctx = factory.createContext())
+      {
+         Assert.assertNotNull(ctx.createConsumer(ctx.createQueue("queue")).receiveNoWait());
+         Assert.assertNull(ctx.createConsumer(ctx.createQueue("queue")).receiveNoWait());
+      }
+
+      factory.close();
```

Comment:
Seems like it would be a good idea for the connection factory to be auto closeable even if the spec doesn't say so.